pkg/setting/setting_session_test.go

```diff
@@ -0,0 +1,43 @@
+package setting
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/log"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type testLogger struct {
+	log.Logger
+	warnCalled  bool
+	warnMessage string
+}
+
+func (stub *testLogger) Warn(testMessage string, ctx ...interface{}) {
+	stub.warnCalled = true
+	stub.warnMessage = testMessage
+}
+func TestSessionSettings(t *testing.T) {
+	Convey("session config", t, func() {
+		skipStaticRootValidation = true
+
+		Convey("Reading session should log error ", func() {
+			var (
```

Comment:
I generally prefer `:=` over var usage. I'm curious, is there a reason why the stub is created outside of the var block?

Reply:
Right! Do we want to document it somewhere? Or better yet - since we using the revive it would be good to create a custom rule for it?

Reply:
> I'm curious, is there a reason why the stub is created outside of the var block?

You can create a type inside a function, but you cannot create a method of that type inside of function i.e. it doesn't compile :)

Or maybe you have different approach in mind?